### PR TITLE
improve the Custom ScmAuthApi Implementation section of docs/auth

### PR DESCRIPTION
**Pull request by dagda1**
## Hey, I just made a Pull Request!

I had a really hard time with the existing section of `Custom ScmAuthApi Implementation` in `docs/auth` so I have detailed the steps I had to complete to get things working.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [X] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by taras** [comment]
@dagda1 nice!

File: docs/auth/index.md [hunk 1]
```diff
@@ -278,3 +289,16 @@ createApiFactory({
     ),
 });
 ```
+
+\_**Warning:** You will need to remove the default `ScmAuth.createDefaultApiFactory();` with this approach.
```
Comment (freben) [inline docs/auth/index.md:293]: Accidental underscore at the start

Comment (freben) [reply docs/auth/index.md:293]: Also maybe go with `**NOTE:** Remember to remove [...]`, warning sounds a bit extreme

File: docs/auth/index.md [hunk 2]
```diff
@@ -278,3 +289,16 @@ createApiFactory({
...
+
+3.  Finally you also need to add and configure another GitHub provider to the `auth-backend` using the provider ID, which in this example is `ghe`:
```
Comment (freben) [inline docs/auth/index.md:295]: The text here is written in a pretty prescriptive way; to me, reading it from top to bottom, it loses its generality (of telling how to make ANY ScmAuthApi) and starts describing the ghe specific case and its nuances pretty closely.

To be clear, it's perfectly fine to make an `ScmAuthApi` that has only the default github provider and the default bitbucket provider in it and nothing else. So, adding a new auth API ref is optional, and adding a provider factory in the backend is also optional. It depends on the circumstances, and the latter is kinda detached from the act of making the `ScmAuthApi` specifically.

Could the text be reworded a bit to clarify this? Maybe the re could be an initial paragraph that only says how to make a custom instance of the scmAuthApiRef (including removing the default one). And then a follow-up paragraph that says that yeah you can also add your entirely own auth refs and inject them into the scm auth api, and you do that by ... etc

Comment (dagda1) [reply docs/auth/index.md]: @freben I have updated the copy, but I actually do not know when you need to add a n entry to `providerFactories` and when you do not so any help on that would be great.

**Review by freben — APPROVED** [review]
Thanks, nice!

**Pull request merged**